kodilla-hibernate/src/test/java/com/kodilla/hibernate/tasklist/dao/TaskListDaoTestSuite.java

```diff
@@ -0,0 +1,32 @@
+package com.kodilla.hibernate.tasklist.dao;
+
+import com.kodilla.hibernate.tasklist.TaskList;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class TaskListDaoTestSuite {
+
+    @Autowired
+    private TaskListDao taskListDao;
+    private static final String LISTNAME = "TEST LIST";
+    private static final String DESCRIPTION = "Test: KOD-17.2";
+
+    @Test
+    public void testFindByListName() {
+        //Given
+        TaskList taskList = new TaskList(LISTNAME, DESCRIPTION);
+        taskListDao.save(taskList);
+
+        //When Then
+        Assert.assertEquals(taskList.getListName(), taskListDao.findByListName(LISTNAME).get(0).getListName());
```

Comment:
1) W tym przypadku nie potrzebujemy stałych w teście, wystarczą zmienne lokalne listName oraz description;
2) Rozdziel sekcję // when oraz then (będzie bardziej elegancko):
`// when`
`List<TaskList> found = taskListdao.findByListName(listName);`
`// then`
` ...`
3) O wiele ładniej jest takie rzeczy robic w następujący sposób (wykracza to poza nasz kurs, ale warto)
`assertThat(found.).extracting(TaskList::getName, TaskList::getDescription).containsExactly(tuple(listName, description));`
Metody `assertThat` oraz `tuple` pochodzą z biblioteki assertj-core (możesz znaleźć na mvnrepository.com i dołączyc do projektu).
4) Myślę, że trzeba jeszcze dodać jedną asercję,a mianowicie:
`assertThat(found.get(0).getId()).isGreaterThan(0);` - to jest sprawdzenie, czy się rzeczywiście zapisał element, bo tylko wtedy id jest nadany przez Hibernate i nie jest zerem.